cachecontrol/caches/redis_cache.py

```diff
@@ -13,11 +13,7 @@ def get(self, key):
         return self.conn.get(key)
 
     def set(self, key, value, expires=None):
-        if not expires:
-            self.conn.set(key, value)
-        else:
-            expires = expires - datetime.utcnow()
-            self.conn.setex(key, int(expires.total_seconds()), value)
+        self.conn.set(key, value, ex=expires)
```

Comment:
if  expires is 0: redis.exceptions.ResponseError: invalid expire time in set

Reply:
First off, my apologies for not having noticed this response until now!

Would it be better to make sure [get_cache_expiry()](https://github.com/ionrock/cachecontrol/pull/206/files/c9c98e26ac290fd31f4cbae2cf9ffc1e1bcf165e#diff-3a5ce26da48f679d11ee0eeb9d57ee58R347) never returns 0 (returning None instead in that case) or do validation in place (something like: ex=expires or None)?  Or both?

Reply:
I've done just the latter - coerce a 0 into a None in set().

We don't win much in putting more logic in get_cache_expiry() unless this logic becomes necessary in other parts of the library.  If a really wrong value is passed, it feels correct to raise an exception.